central/vulnmgmt/vulnerabilityrequest/manager/requestmgr/manager_impl.go

```diff
@@ -640,7 +654,14 @@ func (m *managerImpl) getFixableCVEDeferrals() ([]*storage.VulnerabilityRequest,
 				fixableReqs = append(fixableReqs, request)
 			}
 		} else if expiryType == storage.RequestExpiry_ALL_CVE_FIXABLE {
-			results, err := m.imageCVEs.Search(ctx, fixableQuery)
+
+			var results []search.Result
+			var err error
+			if features.FlattenCVEData.Enabled() {
+				results, err = m.imageCVEsV2.Search(ctx, fixableQuery)
+			} else {
+				results, err = m.imageCVEs.Search(ctx, fixableQuery)
+			}
```

Comment:
Could we make this entire section, both old and new models, simpler by just using the cveView?  
That already returns a field for just the CVE so we wouldn't have to parse the ids.

@charmik-redhat what do you think?

Reply:
@dashrews78 @charmik-redhat  is this the suggestion 
`results, err := m.ImageCVEView.Get(ctx, fixableQuery, centralViews.ReadOptions{})
			if err != nil {
				return nil, errors.Wrapf(err, "could not search image cves for request %q", request.GetId())
			}
			cveSet := set.NewStringSet()
			for _, cveCore := range results {
				cveIds := cveCore.GetCVEIDs()
				for _,cveId := range cveIds{
					cveSet.Add(cveId)	
				}
				
			}
			if cveSet.Cardinality() == len(request.GetCves().GetCves()) {
				fixableReqs = append(fixableReqs, request)
			}`

Reply:
@charmik-redhat @dashrews78 is this the suggestion
```suggestion
results, err := m.ImageCVEView.Get(ctx, fixableQuery, centralViews.ReadOptions{})
			if err != nil {
				return nil, errors.Wrapf(err, "could not search image cves for request %q", request.GetId())
			}
			cveSet := set.NewStringSet()
			for _, cveCore := range results {
				cveIds := cveCore.GetCVEIDs()
				for _, cveId := range cveIds {
					cveSet.Add(cveId)
				}

			}
```

Reply:
Not quite.  If you notice the original is adding the `CVE` field to the set NOT the `CVE ID` field.  `m.ImageCVEView.Get` will return a single instance of the `CVE` field, I believe.  So you may not even need the post loop if we use that function.